src/program/types/ClassBody.js

```diff
@@ -49,6 +49,15 @@ export default class ClassBody extends Node {
 				if (!inFunctionExpression) code.appendLeft(constructor.end, ';');
 			}
 
+			const classFields = [];
+			this.body.forEach(method => {
+				if (method.type === 'FieldDefinition') {
+					classFields.push(`this.${code.slice(method.start, method.end)};`);
+					code.remove(method.start, method.end);
+					return;
+				}
+			});
+
```

Comment:
Maybe it's possible to use `move` here instead of removing the text and inserting it somewhere else?

Reply:
I'm not sure it's possible to do this, since we're creating the constructor as a string, not a magic string. Right now the logic is `collect fields => join fields with newlines => insert into top of constructor`. To do this with `move` (besides converting constructor to magic string), we'd have to loop backwards (so we can always append to the front) or track the end position of the last inserted field.

Reply:
Yeah, it's probably complicated if possible at all. Let's leave it as it currently is and deal with issues once they arise.